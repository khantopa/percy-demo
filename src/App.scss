@import './styles/main.scss';

.App {
  position: relative;
  max-width: 100vw;
  width: 100vw;
  height: 100vh;
  max-height: 100vh;
  overflow-x: hidden;
  background: $primary;  
}

