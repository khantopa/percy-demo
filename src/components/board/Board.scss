@import '../../styles/main.scss';

.play {
  width: 100%;
  height: 100vh;
  padding: 20px 40px;
  box-sizing: border-box;
  &_title {
    text-align: center;
    color: $light;
  }
  &_board {
    position: relative;
    padding: 20px 40px;
    height: 300px;  
    &_players {
      @include grid(2);
      color: $light;
      margin-bottom: 20px;
    }
    &_result {
      text-align: center;
      animation: fadeIn .5s ease;
      color: $light;
      font-size: 20px;
      font-weight: bolder;
    }
  }
  &_button { 
    width: 300px;
    margin: 10px auto;
    @include grid(3);
    grid-gap: 10px;
    &_item {
     @extend .icon_button;
    }
  }
  &_text {
    text-align: center;
    color: black;
    font-size: bolder;
  }
  &_run {
    @include grid(1);
    &_button {
      @extend .icon_button;
    }
  }
  &_footer {
    width: 100%;
    margin-top: 20px; 
    box-sizing: border-box;
    @include grid(2);
    grid-gap: 20px;
    &_button {
      @extend .button;
      width: 200px;
      background: $dark;
      &.reset {
        background: $secondary;
      }
    }
  }
}

@media (max-width: $mobile-width) {
  .play {
    width: 100vw;
    padding: 2px;
    &_board {
      padding: 20px 0;
      height: 230px;
    }
    &_footer {
      margin-top: 40px;
      @include grid(1);
      grid-gap: 10px;
    }
  }
}