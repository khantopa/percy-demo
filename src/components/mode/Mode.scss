@import '../../styles/main.scss';

.mode {
  width: 300px;
  height: 300px;
  box-sizing: border-box;
  @include absPosition(50%, 25%);
  transform: translateX(-50%);
  text-align: center;
  padding: 10px;
  &_head {
    margin-bottom: 20px;
    color: white;
  }
  &_group {
    margin-top: 80px; 
    color: white;
    &_button {
      width: 100%;
      margin: 40px auto;
      @extend .button;
    }
  }
}


@media screen and (max-width: $mobile-width) {
  .mode {
    padding: 20px;
    width: 100vw;
    height: 300px;
  }
}