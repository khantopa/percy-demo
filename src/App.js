import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import Mode from './components/mode/Mode';
import Board from './components/board/Board';

import './App.scss';


function App() {
  return (
   <div className="App">
    <Router>
      <Route exact path="/" component={ Mode } />
      <Route path="/play" component= { Board }/>
    </Router>
   </div>
  );
}

export default App;
