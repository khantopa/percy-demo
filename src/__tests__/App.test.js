import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import App from '../App';
import Mode from '../components/Mode';
import Board from '../components/Board';

let wrapper, router;

beforeEach(() => {
  wrapper = shallow(<App/> );
  router = wrapper.find(Router);
});

describe('App', () => {

  it('should <App> defined', () => {
    expect(wrapper).toBeDefined();
  });
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should have Router ', () => {
    expect(wrapper.find(Router)).toHaveLength(1);
  });
  
  it('Should have Router ', () => {
    expect(router.find(Route)).toHaveLength(2);
  });

  it('Should renders correct routes', () => {
    const pathMap = router.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.component;
      return pathMap;
    }, {});  
    expect(pathMap['/']).toBe(Mode);
    expect(pathMap['/play']).toBe(Board);
  });

});

