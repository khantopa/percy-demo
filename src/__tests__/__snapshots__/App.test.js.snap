// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/play"
    />
  </BrowserRouter>
</div>
`;
